sources/osg/Stack.js

```diff
@@ -1,22 +1,23 @@
 'use strict';
 var Stack = function() {
-    this.globalDefault = undefined;
-    this.lastApplied = undefined;
-    this.changed = false;
+    this._globalDefault = undefined;
+    this._lastApplied = undefined;
+    this._changed = false;
 
-    this.values = [];
-    this.back = undefined;
+    this._values = [];
+    this._back = undefined;
 };
 
 Stack.prototype = {
     push: function(value) {
-        this.values.push(value);
-        this.back = value;
+        this._values.push(value);
+        this._back = value;
     },
     pop: function() {
-        var values = this.values;
+        var values = this._values;
+        if (!values.length) return undefined;
         var value = values.pop();
-        this.back = values[values.length - 1];
+        this._back = values.length ? values[values.length - 1] : undefined;
```

Comment:
Not sure but maybe it would make sense to have the same api than the pool and maybe transform this stack to StackPool